src/modules/airline/index.ts

```diff
@@ -0,0 +1,232 @@
+    options: {
+      allowNumerics?: boolean;
+      allowVisuallySimilarCharacters?: boolean;
+    } = {}
+  ): string {
+    const { allowNumerics = false, allowVisuallySimilarCharacters = false } =
+      options;
+    const excludedChars = [];
+    if (!allowNumerics) {
+      excludedChars.push(...numerics);
+    }
+
+    if (!allowVisuallySimilarCharacters) {
+      excludedChars.push(...visuallySimilarCharacters);
+    }
+
+    return this.faker.string.alphanumeric({
+      length: 6,
+      casing: 'upper',
+      exclude: excludedChars,
+    });
+  }
+
+  /**
+   * Generates a random seat.
+   *
+   * @param options The options to use. Defaults to `{}`.
+   * @param options.aircraftType The aircraft type. Can be one of `narrowbody`, `regional`, `widebody`. Defaults to `narrowbody`.
+   *
+   * @example
+   * faker.airline.seat() // '22C'
+   * faker.airline.seat({ aircraftType: 'regional' }) // '7A'
+   * faker.airline.seat({ aircraftType: 'widebody' }) // '42K'
+   *
+   * @since 8.0.0
+   */
+  seat(
+    options: {
+      aircraftType?: AircraftType;
+    } = {}
+  ): string {
+    const { aircraftType = Aircraft.Narrowbody } = options;
+    const maxRow = aircraftTypeMaxRows[aircraftType];
+    const allowedSeats = aircraftTypeSeats[aircraftType];
+    const row = this.faker.number.int({ min: 1, max: maxRow });
+    const seat = this.faker.helpers.arrayElement(allowedSeats);
+    return `${row}${seat}`;
+  }
+
+  /**
+   * Returns a random aircraft type.
+   *
+   * @example
+   * faker.airline.aircraftType() // 'narrowbody'
+   *
+   * @since 8.0.0
+   */
+  aircraftType(): AircraftType {
+    return this.faker.helpers.objectValue(Aircraft);
+  }
+
+  /**
+   * Returns a random flight number. Flight numbers are always 1 to 4 digits long. Sometimes they are
+   * used without leading zeros (e.g.: American Airlines flight 425) and sometimes with leading
+   * zeros, often with the airline code prepended (e.g.: AA0425).
+   *
+   * @param options The options to use. Defaults to `{}`.
+   * @param options.length The number or range of digits to generate. Defaults to `{ min: 1, max: 4 }`.
+   * @param options.addLeadingZeros Whether to pad the flight number up to 4 digits with leading zeros. Defaults to `false`.
+   *
+   * @example
+   * faker.airline.flightNumber() // '2405'
+   * faker.airline.flightNumber({ addLeadingZeros: true }) // '0249'
+   * faker.airline.flightNumber({ addLeadingZeros: true, length: 2 }) // '0042'
+   * faker.airline.flightNumber({ addLeadingZeros: true, length: { min: 2, max: 3} }) // '0624'
+   * faker.airline.flightNumber({ length: 3 }) // '425'
+   * faker.airline.flightNumber({ length: { min: 2, max: 3 } }) // '84'
+   *
+   * @since 8.0.0
+   */
+  flightNumber(
+    options: {
+      length?: number | { min: number; max: number };
+      addLeadingZeros?: boolean;
+    } = {}
+  ): string {
+    const { length = { min: 1, max: 4 } } = options;
+    const { addLeadingZeros = false } = options;
```

Comment:
You can merge these into a single statement:

```suggestion
    const { length = { min: 1, max: 4 }, addLeadingZeros = false } = options;
```

(Might require formatting)

Reply:
Thanks for catching that!